### DISCUSSION TITLE
View generate SQL

### DISCUSSION THREAD

**Discussion by glockjt** [supabase/supabase#3382] [Questions] [answered]
Is there a way to view the generated query? Like knex has `.debug()` I cannot find anywhere in the documentation to debug or print out the generated SQL.

**Comment by silentworks** [comment]
There is no way to do it with the supabase-js library. The queries are all handled on the server itself with postgrest. You can checkout postgrest documentation to get an idea of how the queries are generated https://postgrest.org/en/v8.0/api.html#

**Reply by steve-chavez** [reply]
>  Like knex has .debug()

Also note that knex is purely a backend library while `supabase-js`/`postgrest-js` are client libraries(can be used on React, Vue, etc). The way to view the generated queries is likely going to be on the Supabase UI Logs(this is a future enhancement).

**Reply by eMeRiKa13** [reply]
Supabase UI Logs will be an awesome improvement!

**Comment by Ziinc** [comment, marked as answer]
Hi @glockjt @eMeRiKa13 , additional logging features have been made available in the Dashboard, and documentation on its features and capabilities are available [here](https://supabase.com/docs/guides/platform/logs).

**Reply by glockjt** [reply]
This is great, thank you for following up!

**Reply by eMeRiKa13** [reply]
Awesome thanks you!

**Reply by e-lobo** [reply]
which column shows the sql? i do not see statement column

<img width="661" alt="image" src="https://github.com/supabase/supabase/assets/41537985/37b7da95-d258-4995-819d-7358b3459bb0">

**Reply by Ziinc** [reply]
The executed statement will usually be in the event message
